analyzer/queries/queries.go

```diff
@@ -367,6 +377,47 @@ const (
       runtime = $1 AND
       token_address = $2`
 
+	//nolint:gosec // thinks this is an authentication "token"
+	RuntimeEVMTokenBalanceAnalysisStale = `
+	SELECT
+		evm_token_balance_analysis.token_address,
+		evm_token_balance_analysis.account_address,
+		evm_token_balance_analysis.last_mutate_round,
+		evm_tokens.token_type,
+		evm_token_balances.balance,
+		token_address_preimage.context_identifier,
+		token_address_preimage.context_version,
+		token_address_preimage.address_data,
+		account_address_preimage.context_identifier,
+		account_address_preimage.context_version,
+		account_address_preimage.address_data
+	FROM chain.evm_token_balance_analysis
+	JOIN chain.evm_token_analysis USING (runtime, token_address)
+	LEFT JOIN chain.evm_tokens USING (runtime, token_address)
+	LEFT JOIN chain.evm_token_balances USING (runtime, token_address, account_address)
+	LEFT JOIN chain.address_preimages AS token_address_preimage ON
+		token_address_preimage.address = evm_token_balance_analysis.token_address
+	LEFT JOIN chain.address_preimages AS account_address_preimage ON
+		account_address_preimage.address = evm_token_balance_analysis.account_address
+	WHERE
+		evm_token_balance_analysis.runtime = $1 AND
+		(
+			evm_token_balance_analysis.last_download_round IS NULL OR
+			evm_token_balance_analysis.last_mutate_round > evm_token_balance_analysis.last_download_round
+		) AND
+		evm_token_analysis.last_download_round IS NOT NULL
+	LIMIT $2`
+
+	//nolint:gosec // thinks this is an authentication "token"
+	RuntimeEVMTokenBalanceAnalysisUpdate = `
+	UPDATE chain.evm_token_balance_analysis
```

Comment:
too much of a leap imo. the fact that there's no other balance in the EVM seems coincidental. we were pretty close to having the EVM manage a native balance of its own.

and I think this https://github.com/oasisprotocol/oasis-indexer/commit/eadefeee0e7a57b4b17ff0dbb695d8613447e23f really took our naming in a direction... I wanted the "token" balances table to be something that's only distinct from the native balance. https://www.coingecko.com/learn/crypto-coins-vs-tokens, i.e. not distinct based on what execution platform it's implemented on. so cipher OAS-20 contracts would also create "tokens," the balances of which we could track in this table. that's why the keys are oasis addresses and we go through the preimage lookup as needed.

the name shuffling in #288 changed "token balances" to "evm balances," which brings us farther from that vision, cosmetically. but I'm still going to push for homogeneity where we can have it, and if we ever work on indexing cipher tokens, I'm going to propose that we generalize evm_token_balance to token_balance. and I'm going to oppose changing to evm_balance at this time.

Reply:
Aha, good thinking about Cipher and non-evm tokens. I wasn't aware of it. :+1:  I'm on board. I don't feel bad about having "hardcoded" for evm though (as we've also done in a lot of our APIs); plans/requirements change a lot over time, so overgeneralization is just as real of a "danger" as oversimplification.

Not a fan of trying to establish a coin vs token distinction in our naming though. It's not a prevalent distinction in common usage. Also philosophically, is emerald ROSE a coin because it's native to the runtime / an L2 chain, or is it a token because it's managed by a runtime, and only consensus ROSE is a true coin? (Rhetorical question, just to say I think both answers are defensible.)